### ISSUE TITLE
openshift 4.8.0-0.nightly-2021-03-04-003632 IPI failed

### ISSUE THREAD

**Issue by zhouhao3** [openshift/installer#4857]
# Version

```console
$ openshift-baremetal-install version
openshift-baremetal-install 4.8.0-0.nightly-2021-03-04-003632
built from commit 54ae8bcf431e8e6d2b12445c3ae80c9b074cc45b
release image quay.io/openshift-release-dev/ocp-release-nightly@sha256:3c9f9135b28d942a89332aa661a6ee0fe96a40579ed1e14f70eb3d5e9ecc17c4
```

# Platform:

Baremetal

IPI

# What happened?

When deploying a cluster using IPI, one of the three masters (randomly) has not been deployed yet on node2, and the debug information is missing. 
```
time="2021-04-17T09:17:13Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [31m10s elapsed]"
time="2021-04-17T09:17:23Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [31m20s elapsed]"
time="2021-04-17T09:17:23Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [31m20s elapsed]"
time="2021-04-17T09:17:23Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [31m20s elapsed]"
time="2021-04-17T09:17:33Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [31m30s elapsed]"
time="2021-04-17T09:17:33Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [31m30s elapsed]"
time="2021-04-17T09:17:33Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [31m30s elapsed]"
time="2021-04-17T09:17:43Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [31m40s elapsed]"
time="2021-04-17T09:17:43Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [31m40s elapsed]"
time="2021-04-17T09:17:43Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [31m40s elapsed]"
time="2021-04-17T09:17:53Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [31m50s elapsed]"
time="2021-04-17T09:17:53Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [31m50s elapsed]"
time="2021-04-17T09:17:53Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [31m50s elapsed]"
time="2021-04-17T09:18:03Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [32m0s elapsed]"
time="2021-04-17T09:18:03Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [32m0s elapsed]"
time="2021-04-17T09:18:13Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [32m10s elapsed]"
time="2021-04-17T09:18:13Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [32m10s elapsed]"

The above is a piece of continuous log information, node2 has no relevant information before it has been created.

This makes IPI mistakenly think that it is created. 

When the other two nodes are successfully created and the next step is performed, there will be an error that node2 cannot be found (because the creation is not completed), resulting in IPI error exit.

```
time="2021-04-17T09:33:53Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [47m51s elapsed]"
time="2021-04-17T09:33:53Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [47m51s elapsed]"
time="2021-04-17T09:33:59Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[0]: Creation complete after 47m56s [id=550baf18-2f9d-459e-95b7-3ef9b9bbc4e7]"
time="2021-04-17T09:33:59Z" level=debug msg="module.masters.ironic_node_v1.openshift-master-host[1]: Creation complete after 47m56s [id=1441b701-bc6b-48e1-adad-4c74053b5268]"
time="2021-04-17T09:33:59Z" level=error
time="2021-04-17T09:33:59Z" level=error msg="Error: could not make node available: Get \"http://192.168.30.201:6385/v1/nodes/bac5a243-b198-4d0e-ae06-51f4a048f852\": dial tcp 192.168.30.201:6385: connect: no route to host"
time="2021-04-17T09:33:59Z" level=error
time="2021-04-17T09:33:59Z" level=error msg="  on ../../tmp/openshift-install-366331965/masters/main.tf line 1, in resource \"ironic_node_v1\" \"openshift-master-host\":"
time="2021-04-17T09:33:59Z" level=error msg="   1: resource \"ironic_node_v1\" \"openshift-master-host\" {"
time="2021-04-17T09:33:59Z" level=error
time="2021-04-17T09:33:59Z" level=error
time="2021-04-17T09:33:59Z" level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
time="2021-04-17T09:35:40Z" level=debug msg="OpenShift Installer 4.8.0-0.nightly-2021-03-04-003632"
time="2021-04-17T09:35:40Z" level=debug msg="Built from commit 54ae8bcf431e8e6d2b12445c3ae80c9b074cc45b"
```

# How to reproduce it (as minimally and precisely as possible)?

<!--
Please list the full steps required to reproduce the issue.
-->

```console
openshift-baremetal-install --dir ~/clusterconfigs create manifests
 openshift-baremetal-install --dir ~/clusterconfigs --log-level debug create cluster

```

# Anything else we need to know?

I located this error and found that the version 4.8.0-0.nightly-2021-03-03-220107 will not have this error, from 4.8.0-0.nightly-2021-03-04-003632 This error occurred in the version.

Then I compared the commit difference of openshift-install corresponding to the two versions and found that there was only a gap between two PRs.([PR4702](https://github.com/openshift/installer/pull/4702) and [PR4692](https://github.com/openshift/installer/pull/4692)) So I think the reason may be related to these two PRs.
# References

- enter text here.

**Comment by zhouhao3** [comment]
@stbenjam Can you take a look?

**Comment by zhouhao3** [comment]
The latest survey results are as follows:
1. We tried deploying with two masters and one master respectively, and this error did not appear.
2. By modifying the code, the error message when there is a problem is output:
 ```
DEBUG module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [10m30s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [10m30s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [10m40s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[2]: Still creating... [10m40s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [10m40s elapsed] 
*******************************************************************  
apply error： could not manage: Get "http://192.168.30.201:6385/v1/nodes/285ea1a6-efc9-45eb-9d50-46dc33df6232": dial tcp 192.168.30.201:6385: connect: no route to host
*******************************************************************                                                                                                          
DEBUG module.masters.ironic_node_v1.openshift-master-host[0]: Still creating... [10m50s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [10m50s elapsed] 
DEBUG module.masters.ironic_node_v1.openshift-master-host[1]: Still creating... [11m0s elapsed] 
 ```
 After investigation, the IP at that time is indeed unreachable, because during the installation process, the `apiVIP` in the virtual machine will disappear for a period of time (about one minute). 192.168.30.201 cannot be connected during this time.
 
3. During the deployment using two masters, the `apiVIP` in the virtual machine will disappear for a while, but the installation of IPI will not cause an error.

**Comment by stbenjam** [comment]
Hi @zhouhao3 Thanks for your investigation, we're aware of the problem and looking into some solutions for it. We'll keep you updated when we have some fixes up.

**Comment by zhouhao3** [comment]
Fixed by #4900 and openshift/baremetal-runtimecfg#135.
